universum/modules/reporter.py

```diff
@@ -119,6 +122,8 @@ def report_build_result(self):
 
         if not self.observers:
             self.out.log("Nowhere to report. Skipping...")
+            if self.settings.fail_unsuccessful and not is_successful:
```

Comment:
I think that setting of exit code has to be function of main.py, not reporter.
main.pu can use data from reporter for this, but it has to set the exit code on his own.

Reply:
Otherwise, you will not be able to change something in main.py.
For example, you cannot add any code after reporter, because it will not be run due to exception.
I propose to commit this as is, and refactor in the next PR.